From 1e4e9248ef6bcdd95ec3b91c8c8e94c4587a876b Mon Sep 17 00:00:00 2001
From: Ray Strode <rstrode@redhat.com>
Date: Fri, 3 Jul 2015 13:54:36 -0400
Subject: [PATCH] screenShield: unblank when inserting smartcard

If a user inserts the smartcard when the screen is locked/blanked
we should ask them their pin right away.

At the moment they have to wiggle the mouse or do some other
action to get the screen to unblank.
---
 js/ui/screenShield.js | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/js/ui/screenShield.js b/js/ui/screenShield.js
index 9a64fc32c..bc1a0fba7 100644
--- a/js/ui/screenShield.js
+++ b/js/ui/screenShield.js
@@ -85,8 +85,10 @@ var ScreenShield = class {
         this._smartcardManager = SmartcardManager.getSmartcardManager();
         this._smartcardManager.connect('smartcard-inserted',
                                        (manager, token) => {
-                                           if (this._isLocked && token.UsedToLogin)
+                                           if (this._isLocked && token.UsedToLogin) {
+                                               this._wakeUpScreen();
                                                this._activateDialog();
+                                           }
                                        });
 
         this._oVirtCredentialsManager = OVirt.getOVirtCredentialsManager();
-- 
2.31.1

